qemu/tests/numa_cpu.py

```diff
@@ -0,0 +1,149 @@
+        return numa_cpu_guest
+
+    def numa_cpu_cli():
+        """
+        Get the numa cpus for each node according to the qemu cli
+        """
+        error_context.context("Get the expected cpus in qemu command line", logging.info)
+        numa_cpus = params.objects("guest_numa_cpus")
+        numa_cpu_cli = []
+        tmp = {}
+        for numa_cpu in numa_cpus:
+            numa_cpu_params = params.object_params(numa_cpu)
+            nodeid = numa_cpu_params["numa_cpu_nodeid"]
+            socket = numa_cpu_params.get("numa_cpu_socketid")
+            die = numa_cpu_params.get("numa_cpu_dieid")
+            core = numa_cpu_params.get("numa_cpu_coreid")
+            thread = numa_cpu_params.get("numa_cpu_threadid")
+            cpu_list = convert_cpu_topology_to_index(socket, die, core, thread)
+            if nodeid in tmp.keys():
+                tmp[nodeid] += cpu_list
+            else:
+                tmp[nodeid] = cpu_list
+        for item in sorted(tmp.items(), key=lambda item: item[0]):
+            numa_cpu_cli.append(set(sorted(item[1])))
+        return numa_cpu_cli
+
+    def get_hotpluggable_cpus():
+        """
+        Parse output of "query-hotpluggable-cpus", get the numa cpus for each node.
+        """
+        error_context.context("Get the hotpluggable cpus", logging.info)
+        binded_cpus = []
+        tmp = {}
+        out = vm.monitor.info("hotpluggable-cpus")
+        for vcpu_info in out:
+            vcpus_count = vcpu_info["vcpus-count"]
+            vcpu_info = vcpu_info["props"]
+            nodeid = vcpu_info.get("node-id")
+            socketid = vcpu_info.get("socket-id")
+            dieid = vcpu_info.get("die-id")
+            coreid = vcpu_info.get("core-id")
+            threadid = vcpu_info.get("thread-id")
+            if nodeid is not None:
+                if vm_arch in ('x86_64', 'i386'):
+                    cpu_index = [socketid * socket_weight + dieid * die_weight +
+                                 coreid * core_weight + threadid * thread_weight]
+                elif vm_arch in ('ppc64', 'ppc64le'):
+                    cpu_index = list(range(coreid, coreid + vcpus_count))
+                if nodeid in tmp.keys():
+                    tmp[nodeid] += cpu_index
+                else:
+                    tmp[nodeid] = cpu_index
+        for item in sorted(tmp.items(), key=lambda item: item[0]):
+            binded_cpus.append(set(sorted(item[1])))
+        return binded_cpus
+
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login()
+    os_type = params["os_type"]
+    vm_arch = params["vm_arch_name"]
+
+    vcpu_threads = params.get_numeric('vcpu_threads')
+    if vm_arch in ('x86_64', 'i386'):
+        vcpu_sockets = params.get_numeric('vcpu_sockets')
+        vcpu_dies = params.get_numeric('vcpu_dies')
+        vcpu_cores = params.get_numeric('vcpu_cores')
+
+        socket_weight = vcpu_dies * vcpu_cores * vcpu_threads
```

Comment:
It means the total number of cpus with the same socketid, right? The word 'weight' sounds a bit weird to me.

Reply:
> It means the total number of cpus with the same socketid, right? The word 'weight' sounds a bit weird to me.

Take binary as an example, the weight of the lowest bit is 1, the second bit is 2(2 choices for the first bit), and the third bit is 2(2 choice for first bit )*2(2 choice for second bit), and so on...
It is same here.